plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Core/Instance/Flat.hs

```diff
@@ -0,0 +1,57 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+{-# LANGUAGE LambdaCase           #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Language.UntypedPlutusCore.Core.Instance.Flat where
+
+import           Language.UntypedPlutusCore.Core.Type
+
+import           Language.PlutusCore.Flat
+import           Language.PlutusCore.Universe
+
+import           Data.Word                            (Word8)
+import           Flat
+import           Flat.Decoder
+import           Flat.Encoder
+
+encodeTerm :: Word8 -> Encoding
+encodeTerm = eBits 4
```

Comment:
This is where the magic happens, right? i.e. this is where we do the fixed-size small integer encoding. Worth a comment. In fact, we probably need a Note generally explaining the point of `flat` and how we get a compact encoding from it.

Reply:
Yes. I will add a note and comments specifying the range of values it can use.